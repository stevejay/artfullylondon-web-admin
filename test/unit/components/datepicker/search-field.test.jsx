import React from 'react'
import { shallow } from 'enzyme'
import _ from 'lodash'

import DatepickerSearchField from '_admin/components/datepicker/search-field'

it('should render correctly', () => {
  const wrapper = shallow(
    <DatepickerSearchField
      input={{
        value: '2017/01/01',
        onChange: _.noop
      }}
      meta={{
        touched: false,
        error: null
      }}
      disabled={false}
      minDate='2016/01/01'
      htmlId='some-id'
      dateFormat='YYYY/MM/DD'
      placeholder='The Placeholder'
      showModal={_.noop}
    />
  )

  expect(wrapper).toMatchSnapshot()
})

it('should render correctly when it has no value', () => {
  const wrapper = shallow(
    <DatepickerSearchField
      input={{
        value: null,
        onChange: _.noop
      }}
      meta={{
        touched: false,
        error: null
      }}
      disabled={false}
      minDate='2016/01/01'
      htmlId='some-id'
      dateFormat='YYYY/MM/DD'
      placeholder='The Placeholder'
      showModal={_.noop}
    />
  )

  expect(wrapper).toMatchSnapshot()
})

it('should render correctly when it has an error', () => {
  const wrapper = shallow(
    <DatepickerSearchField
      input={{
        value: '2017/01/01',
        onChange: _.noop
      }}
      meta={{
        touched: true,
        error: 'The Error'
      }}
      disabled={false}
      minDate='2016/01/01'
      htmlId='some-id'
      dateFormat='YYYY/MM/DD'
      placeholder='The Placeholder'
      showModal={_.noop}
    />
  )

  expect(wrapper).toMatchSnapshot()
})
