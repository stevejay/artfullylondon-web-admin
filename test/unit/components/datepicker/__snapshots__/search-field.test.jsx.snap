// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  styleName="container"
>
  <div
    styleName="text-wrapper"
  >
    <Text
      autos={true}
      disabled={false}
      error={false}
      forceSingleLine={true}
      id="some-id"
      maxLength={100}
      onClick={[Function]}
      password={false}
      placeholder=""
      readOnly={true}
      styleName="text"
      value="2017/01/01"
    />
  </div>
  <ResetButton
    disabled={false}
    onClick={[Function]}
  />
</div>
`;

exports[`should render correctly when it has an error 1`] = `
<div
  styleName="container"
>
  <div
    styleName="text-wrapper"
  >
    <Text
      autos={true}
      disabled={false}
      error={true}
      forceSingleLine={true}
      id="some-id"
      maxLength={100}
      onClick={[Function]}
      password={false}
      placeholder=""
      readOnly={true}
      styleName="text"
      value="2017/01/01"
    />
  </div>
  <ResetButton
    disabled={false}
    onClick={[Function]}
  />
</div>
`;

exports[`should render correctly when it has no value 1`] = `
<div
  styleName="container"
>
  <div
    styleName="text-wrapper"
  >
    <Text
      autos={true}
      disabled={false}
      error={false}
      forceSingleLine={true}
      id="some-id"
      maxLength={100}
      onClick={[Function]}
      password={false}
      placeholder=""
      readOnly={true}
      styleName="text"
      value="Select a date..."
    />
  </div>
  <ResetButton
    disabled={false}
    onClick={[Function]}
  />
</div>
`;
